### ISSUE TITLE
`confloat`: TypeError: Interval() takes no arguments

### ISSUE THREAD

**Issue by EMalagolihpa** [pydantic/pydantic#6418]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2 installed directly from the `main` branch, or equivalent

### Description

*Python 3.9.7, annotated_types==0.5.0*
`confloat` with arguments (for example `strict=True`), show conflicts with `annotated_types.Interval` which does not take arguments.

*Python 3.10.11, annotated_types==0.5.0* 
No error.

Python version incompatibility?

### Example Code

```Python
from pydantic import BaseModel
from pydantic import confloat

class Model(BaseModel):
    a: confloat(strict=True)

m = Model(a=3.0)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.0
        pydantic-core version: 2.0.1 release build profile
                 install path: /home/hpa/anaconda3/envs/dinema_io/lib/python3.9/site-packages/pydantic
               python version: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]
                     platform: Linux-5.15.0-76-generic-x86_64-with-glibc2.31
     optional deps. installed: ['typing-extensions']
```

Selected Assignee: @lig

**Comment by lig** [comment]
@EMalagolihpa I've tested your example on Python 3.9.17 and I get no error.

Could you, please, provide an example that fails and the full output. Thank you!

**Comment by EMalagolihpa** [comment]
@lig The following code

```
from pydantic import confloat
confloat(strict=True)
```
gives me:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hpa/anaconda3/envs/test/lib/python3.9/site-packages/pydantic/types.py", line 213, in confloat
    annotated_types.Interval(gt=gt, ge=ge, lt=lt, le=le),
TypeError: Interval() takes no arguments
```

The error, seems to be in `annotated_types.Interval`. In fact, an even simpler example, if you create a python=3.9.7. env, and you install `annotated_types==0.5.0` and try:

```
import annotated_types
annotated_types.Interval(gt=0)
```
gives same error:
```
TypeError: Interval() takes no arguments
```

If I install the same version of `annotated_types` in a Python=3.10.11 env, no errors.

**Comment by lig** [comment]
Sorry, I still cannot reproduce this.

```shell
~/tmp/annotated-types-error 
❯ virtualenv3.9 .venv3.9
created virtual environment CPython3.9.17.final.0-64 in 231ms
  creator CPython3Posix(dest=/home/lig/tmp/annotated-types-error/.venv3.9, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/lig/.local/share/virtualenv)
    added seed packages: pip==23.1.2, setuptools==67.8.0, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

~/tmp/annotated-types-error 
❯ . ./.venv3.9/bin/activate.fish 

~/tmp/annotated-types-error via 🐍 v3.9.17 (.venv3.9) 
❯ pip install 'annotated_types==0.5.0'
Collecting annotated_types==0.5.0
  Using cached annotated_types-0.5.0-py3-none-any.whl (11 kB)
Installing collected packages: annotated_types
Successfully installed annotated_types-0.5.0

~/tmp/annotated-types-error via 🐍 v3.9.17 (.venv3.9) 
❯ python
Python 3.9.17 (main, Jun  8 2023, 00:00:00) 
[GCC 13.1.1 20230511 (Red Hat 13.1.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
```
```py
>>> import annotated_types
>>> annotated_types.Interval(gt=0)
Interval(gt=0, ge=None, lt=None, le=None)
```

Please, try creating an empty virtual environment and ensure that you are using it. Maybe, you've reused a directory containing files of another environment for creating a new one.

**Comment by EMalagolihpa** [comment]
```
python -m venv venv
source venv/bin/activate
pip install annotated_types==0.5.0
```
and then
```
import annotated_types
annotated_types.Interval(gt=0)
```

Please try with python=3.9.7.

**Comment by lig** [comment]
Thanks @Kludex for reproducing this. It turns out. The error could be reproduced on exactly Python 3.9.7. It doesn't reproduce on the latest Python 3.9.

**Comment by AlexVndnblcke** [comment]
@EMalagolihpa I ran into the same issue with python 3.9.7 and it is effectively gone in 3.9.8.

This is due the the fact that `Interval` inherits from a class that is a sub-type of `typing.Protocol`. I recall this from a previous issue couple months back, but can't seem to find it back anywhere.

For `annotated-types` I've noticed that version 0.4.0 didn't work with `typing.Protocol` so I tried to use that version which was successful.

So, as a workaround I fell back to `annotated_types==0.4.0` which seems to work with python 3.9.7.
